packages/core/core/src/middlewares/security.ts

```diff
@@ -71,12 +71,24 @@ export const security: Common.MiddlewareFactory<Config> =
       });
     }
 
-    if (ctx.method === 'GET' && ['/admin'].some((str) => ctx.path.startsWith(str))) {
+    /**
+     * These are for vite's watch mode so it can accurately
+     * connect to the HMR websocket & reconnect on failure
+     * or when the server restarts.
+     *
+     * It only applies in development, and only on GET requests
+     * that are part of the admin route.
+     */
+    if (
+      process.env.NODE_ENV === 'development' &&
+      ctx.method === 'GET' &&
+      ['/admin'].some((str) => ctx.path.startsWith(str))
+    ) {
```

Comment:
@innerdvations & @alexandrebodin this is the bit specifically i'd like to get you to view :)

Reply:
So this block was originally added (prior to this PR) for vite on develop mode? If that's the case it should be fine, as long as nothing is relying on this in production.

Reply:
shouldn't we only add http: directive on development, but add the rest on production?

Reply:
If Vite is used in production it just builds the files & then the node app looks to the build folder to serve everything, so it shouldn't need any of this 👍🏼  in reality, we _probably_ should have added this additional check when we first introduced vite.